RNTester/js/AccessibilityExample.js

```diff
@@ -133,6 +143,279 @@ class AccessibilityExample extends React.Component {
+    let buttonTitle = this.state.isEnabled
+      ? 'Disable selection'
+      : 'Enable selection';
+
+    return (
+      <View style={{flex: 1, flexDirection: 'row'}}>
+        <TouchableOpacity
+          ref={this.selectableElement}
+          accessible={true}
+          onPress={() => {
+            this.setState({
+              isSelected: !this.state.isSelected,
+            });
+
+            if (Platform.OS === 'android') {
+              UIManager.sendAccessibilityEvent(
+                findNodeHandle(this.selectableElement.current),
+                UIManager.AccessibilityEventTypes.typeViewClicked,
+              );
+            }
+          }}
+          accessibilityLabel="element 21"
+          accessibilityStates={accessibilityStates}
+          accessibilityHint={accessibilityHint}>
+          <Text>Selectable element example</Text>
+        </TouchableOpacity>
+        <Button
+          onPress={() => {
+            this.setState({
+              isEnabled: !this.state.isEnabled,
+            });
+          }}
+          title={buttonTitle}
+        />
+      </View>
+    );
+  }
+}
+
+class StateOnOffExample extends React.Component {
+  state = {
+    elementState: 'on',
+  };
+
+  _onElementPress = () => {
+    const elementState = this.state.elementState === 'on' ? 'off' : 'on';
+
+    this.setState({
+      elementState: elementState,
+    });
+
+    if (Platform.OS === 'android') {
+      UIManager.sendAccessibilityEvent(
+        findNodeHandle(this),
+        UIManager.AccessibilityEventTypes.typeViewClicked,
+      );
+    }
+  };
+
+  render() {
+    return (
+      <TouchableOpacity
+        onPress={this._onElementPress}
+        accessibilityLabel="element 18"
+        accessibilityStates={[this.state.elementState]}
+        accessibilityHint="click me to change state">
+        <Text>State on/off example</Text>
+      </TouchableOpacity>
+    );
+  }
+}
+
+class ExpandableElementExample extends React.Component {
+  state = {
+    expandState: 'collapsed',
+  };
+
+  _onElementPress = () => {
+    const expandState =
+      this.state.expandState === 'collapsed' ? 'expanded' : 'collapsed';
+
+    this.setState({
+      expandState: expandState,
+    });
+
+    if (Platform.OS === 'android') {
+      UIManager.sendAccessibilityEvent(
+        findNodeHandle(this),
+        UIManager.AccessibilityEventTypes.typeViewClicked,
+      );
+    }
+  };
+
+  render() {
+    return (
+      <TouchableOpacity
+        onPress={this._onElementPress}
+        accessibilityLabel="element 20"
+        accessibilityStates={[this.state.expandState]}
+        accessibilityHint="click me to change state">
+        <Text>Expandable element example</Text>
+      </TouchableOpacity>
+    );
+  }
+}
+
+class AccessibilityRoleAndStateExample extends React.Component<{}> {
+  render() {
+    return (
+      <View>
+        <View
+          accessibilityLabel="element 1"
+          accessibilityRole="alert"
+          accessible={true}>
```

Comment:
@marcmulcahy apologies if you've explained this in the past.

What would happen if the developer supplied `accessibleLabel` and `accessibleRole`, but set `accessible` to false? Is this a necessary orthogonal toggle? I can't think if a use case where one would supply a label and a role but not want the element to be "accessible".

Reply:
@jessebeach I don't think this is a concern, as I think accessible=false would hide the component from accessibility services, so the label, roles, and states would be ignored.